Include/cpython/object.h

```diff
@@ -45,61 +45,14 @@ typedef struct _Py_Identifier {
 #define _Py_static_string(varname, value)  static _Py_Identifier varname = _Py_static_string_init(value)
 #define _Py_IDENTIFIER(varname) _Py_static_string(PyId_##varname, #varname)
 
-/* buffer interface */
-typedef struct bufferinfo {
-    void *buf;
-    PyObject *obj;        /* owned reference */
-    Py_ssize_t len;
-    Py_ssize_t itemsize;  /* This is Py_ssize_t so it can be
-                             pointed to by strides in simple case.*/
-    int readonly;
-    int ndim;
-    char *format;
-    Py_ssize_t *shape;
-    Py_ssize_t *strides;
-    Py_ssize_t *suboffsets;
-    void *internal;
-} Py_buffer;
-
+/* Py_buffer is defined in buffer.h */
+typedef struct bufferinfo Py_buffer;
```

Comment:
This shouldn't be necessary, the API should be included from Python.h.

Reply:
There is a chicken and egg problem. ``buffer.h`` must be included after ``object.h`` has declared ``PyObject``. It's now included twice, one time by ``Python.h`` for public API and one time by ``cpython/object.h`` for internal use.